src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java

```diff
@@ -126,6 +127,47 @@ private Optional<InterfaceTypeDefinition> getInterfaceTypeDefinition(TypeDefinit
         return Optional.empty();
     }
 
+    public List<String> getTypesImplementingInterface(@NotNull PsiElement psiElement, @NotNull String interfaceName) {
+        TypeDefinitionRegistry registry = getRegistry(psiElement);
+
+        // Check if the parentType is actually an interface
+        Optional<InterfaceTypeDefinition> interfaceType = getInterfaceTypeDefinition(registry, interfaceName);
+        if (!interfaceType.isPresent()) {
+            return new ArrayList<>(); // Not an interface
+        }
+
+        List<String> implementingTypes = new ArrayList<>();
```

Comment:
Another minor thing: Perhaps use `Set` here to avoid all possible duplication and just convert to list on return? (Kind of the opposite guidance of avoiding Optional, I suppose...)